qemu/tests/migration_multi_host_auto_converge.py

```diff
@@ -0,0 +1,239 @@
+
+    The test procedure:
+    1) starts vm on master host.
+    2) query migrate capabilities
+    3) a. Stress guest and ensure migration could never finish without
+          auto-converge, set different values for x-cpu-throttle-initial
+          and x-cpu-throttle-increment, during migration check cpu
+          throttling percentage will start from $x-cpu-throttle-initial
+          and increase by $x-cpu-throttle-increment, and this process
+          continues until migration completes or we reach 99% throttled.
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    mig_protocol = params.get("mig_protocol", "tcp")
+    mig_type = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        mig_type = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        mig_type = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        mig_type = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(mig_type, MigrationBase):
+
+        """
+        multihost migration test
+        """
+
+        def __init__(self, test, params, env):
+
+            super(TestMultihostMigration, self).__init__(test, params, env)
+            self.srchost = self.params.get("hosts")[0]
+            self.dsthost = self.params.get("hosts")[1]
+            super(TestMultihostMigration, self).__setup__(test, params, env,
+                                                          self.srchost,
+                                                          self.dsthost)
+            self.need_stress = params.get("need_stress") == "yes"
+            self.need_cleanup = self.need_stress
+            self.sub_type = params.get("sub_type")
+            self.sub_test = params.objects("sub_test")
+            for i in range(1, len(self.sub_test)):
+                if self.sub_test[i].strip() == "enable":
+                    self.sub_test[i] = True
+                else:
+                    self.sub_test[i] = False
+            if self.sub_type == "before_migrate_capability":
+                self.before_migration = self.before_migration_capability
+                self.post_migration = self.post_migration_capability
+
+        def set_auto_converge(self):
+
+            """
+            enable/disable auto-converge
+            """
+
+            for i in range(1, len(self.sub_test)):
+                self.set_migration_capability(self.sub_test[i],
+                                              "auto-converge")
+                self.capabilitys.append("auto-converge")
+                self.capabilitys_state.append(self.sub_test[i])
+                self.get_migration_capability(len(self.capabilitys) - 1)
+                self.capabilitys.pop()
+                self.capabilitys_state.pop()
+
+        @error.context_aware
+        def install_stressapptest(self):
```

Comment:
Suggest you put stressapptest part in the framework of migration, it can be used for both xbzrle an auto converge, also for other sub features of live migration.